website/src/components/AdminArea.tsx

```diff
@@ -0,0 +1,19 @@
+import { useRouter } from "next/router";
+import { useSession } from "next-auth/react";
+import { ReactNode, useEffect } from "react";
+
+export const AdminArea = ({ children }: { children: ReactNode }) => {
+  const router = useRouter();
+  const { data: session, status } = useSession();
+
+  useEffect(() => {
+    if (status === "loading") {
+      return;
+    }
+    if (session?.user.role === "admin") {
+      return;
+    }
+    router.push("/");
+  }, [router, session, status]);
```

Comment:
this use effect can be simplified into 
```ts
if (session?.user.role !== "admin")
   router.push("/")
```

Reply:
> this use effect can be simplified into
> 
> ```ts
> if (session?.user.role !== "admin")
>    router.push("/")
> ```

This will rediect if session is null (when the status is loading or unauthenticated)
![image](https://user-images.githubusercontent.com/33456881/216533839-78170a6f-eaee-4e01-a8e6-ff4e76a67a8b.png)

Reply:
oh I see! makes sense